doc/source/data/getting-started.rst

```diff
@@ -4,113 +4,144 @@ Getting Started
-:meth:`~ray.data.Dataset.iter_batches` or
-:meth:`~ray.data.Dataset.iter_rows`.
-This does not incur a copy, since the blocks of the Dataset are passed by reference
-as Ray objects:
-
-.. literalinclude:: ./doc_code/quick_start.py
-   :language: python
-   :start-after: __data_access_begin__
-   :end-before: __data_access_end__
-
-Datasets can be split up into disjoint sub-datasets.
-Locality-aware splitting is supported if you pass in a list of actor handles to the
-:meth:`~ray.data.Dataset.split` function along with the number of desired
-splits.
-This is a common pattern useful for loading and splitting data between distributed
-training actors:
-
-.. literalinclude:: ./doc_code/quick_start.py
-   :language: python
-   :start-after: __dataset_split_begin__
-   :end-before: __dataset_split_end__
-
-See the :ref:`Consuming Datasets guide <consuming_datasets>` for an in-depth guide
-on accessing and exchanging datasets.
+Apply :ref:`user-defined functions <transform_datasets_writing_udfs>` (UDFs) to
+transform datasets. Ray executes transformations in parallel for performance at scale.
+
+.. testcode::
+
+    import pandas as pd
+
+    # Find rows with spepal length < 5.5 and petal length > 3.5.
+    def transform_batch(df: pd.DataFrame) -> pd.DataFrame:
+        return df[(df["sepal length (cm)"] < 5.5) & (df["petal length (cm)"] > 3.5)]
+
+    transformed_dataset = dataset.map_batches(transform_batch)
+    print(transformed_dataset)
+
+.. testoutput::
+
+    MapBatches
+    +- Dataset(num_blocks=1, num_rows=150, schema={sepal length (cm): double, sepal width (cm): double, petal length (cm): double, petal width (cm): double, target: int64})
+
+To learn more about transforming datasets, read
+:ref:`Transforming Datasets <transforming_datasets>`.
+
+Consume the dataset
+-------------------
+
+Pass datasets to Ray tasks or actors, and access records with methods like
+:meth:`~ray.data.Dataset.iter_batches`.
+
+.. tabbed:: Local
+
+    .. testcode::
+
+        batches = transformed_dataset.iter_batches(batch_size=8)
+        print(next(iter(batches)))
+
+   .. testoutput::
+        :options: +NORMALIZE_WHITESPACE
+
+           sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  target
+        0                5.2               2.7                3.9               1.4       1
+        1                5.4               3.0                4.5               1.5       1
+        2                4.9               2.5                4.5               1.7       2
+
+.. tabbed:: Tasks
+
+   .. testcode::
+
+        @ray.remote
+        def consume(dataset: ray.data.Dataset) -> int:
+            num_batches = 0
+            for batch in dataset.iter_batches(batch_size=8):
+                num_batches += 1
+            return num_batches
+
+        ray.get(consume.remote(transformed_dataset))
+
+.. tabbed:: Actors
+
+    .. testcode::
+
+        @ray.remote
+        class Worker:
```

Comment:
I think we have established a practice to move code into separate/runnable code file, and reference the code block in .rst file. Any reason not to follow that?

Reply:
We shouldn't use `literalinclude` to illustrate object representations because `literalinclude` doesn't test outputs. 

For example, in our current quick start code, the schema code isn't tested:
```
ds.schema()
# -> sepal.length: double
# -> sepal.width: double
# -> petal.length: double
# -> petal.width: double
# -> variety: string
```

In general, we're going to move away from `literalinclude`. See https://github.com/ray-project/ray/pull/31252.

Reply:
@jianoaix our CI can now test code snippets directly so no need to move to a separate file anymore.